### PR TITLE
Upgrade Misk's 3rd party dependencies

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: misk/src/test/kotlin/misk/web/JettyServiceMetricsTest.kt [hunk 1]
```diff
@@ -97,8 +97,8 @@ internal class JettyServiceMetricsTest {
     val metrics = adapter.fromJson(response.body?.string()!!)!!
     assertThat(metrics.queuedJobs).isEqualTo(0.0)
     assertThat(metrics.size).isEqualTo(10.0)
-    assertThat(metrics.utilization).isCloseTo(0.5, Offset.offset(0.2))
-    assertThat(metrics.utilization_max).isCloseTo(0.5, Offset.offset(0.2))
+    assertThat(metrics.utilization).isCloseTo(0.5, Offset.offset(0.35))
+    assertThat(metrics.utilization_max).isCloseTo(0.5, Offset.offset(0.35))
```
Comment (swankjesse) [inline misk/src/test/kotlin/misk/web/JettyServiceMetricsTest.kt]: I tried 0.3 but floating point math hates 0.3.

```
JettyServiceMetricsTest > threadPoolMetrics() FAILED
    java.lang.AssertionError:
    Expecting:
      <0.8>
    to be close to:
      <0.5>
    by less than <0.3> but difference was <0.30000000000000004>.
    (a difference of exactly <0.3> being considered valid)
        at misk.web.JettyServiceMetricsTest.threadPoolMetrics(JettyServiceMetricsTest.kt:100)
```

File: build.gradle [hunk 2]
```diff
@@ -110,7 +110,7 @@ subprojects { subProject ->
     rootProject.getTasksByName("testShardNonHibernate", false).first().dependsOn.add(test)
   }
 
-  if (!name.startsWith("samples:")) {
+  if (!name.startsWith("samples")) {
```
Comment (swankjesse) [inline build.gradle]: the samples/ directory itself was tripping up the plugin

File: misk/src/test/kotlin/misk/web/JettyServiceMetricsTest.kt (hunk 1 above)
Comment (zhxnlai) [reply misk/src/test/kotlin/misk/web/JettyServiceMetricsTest.kt]: should `isCloseTo` be more lenient?

File: misk-metrics-digester/src/test/kotlin/misk/metrics/digester/NaN.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package misk.metrics.digester
+
+import org.assertj.core.api.AbstractDoubleAssert
+
+/**
+ * AssertJ 3.15 broke NaN: https://github.com/joel-costigliola/assertj-core/issues/1783
+ *
+ * TODO: replace with isEqualTo() when AssertJ 3.16 is released.
+ */
+internal fun AbstractDoubleAssert<*>.isEqualToHonorNan(expected: Double) {
```
Comment (zhxnlai) [inline misk-metrics-digester/src/test/kotlin/misk/metrics/digester/NaN.kt]: I feel like `isEqualToHonorNaN` is more readable

**Pull request merged**